### PR TITLE
Cowswap332/environment metadata

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Part of https://github.com/cowprotocol/cowswap/issues/332

Follow up to/waterfalls into https://github.com/cowprotocol/cow-sdk/pull/31

~New metadata `environment`~

New appData optional field named `environment`

# Testing

- Unit tests
- Tested on CowSwap locally using `yalc`

### CODE REVIEW

File: src/utils/appData.spec.ts [hunk 1]
```diff
@@ -146,3 +146,12 @@ test('Invalid: quote metadata - wrong type', async () => {
 
   expect(validation.result).toBeFalsy()
 })
+
+test('Valid: environment metadata', async () => {
+  const document = {
+    ...BASE_DOCUMENT,
+    metadata: { environment: { version: '0.1.0', name: 'prod' } },
```
Comment (ramirotw) [inline src/utils/appData.spec.ts:153]: shouldn't this be `production` as it is in the schema?

> "examples": ["production", "development", "staging", "ens"],

Comment (alfetopito) [reply src/utils/appData.spec.ts:153]: That's just an example. The schema is not meant to enforce the contents.

**Review by W3stside — APPROVED** [review]

**Review by ramirotw — APPROVED** [review]

File: src/api/cow-subgraph/graphql.ts [hunk 2]
```diff
@@ -2641,6 +2663,135 @@ export enum UniswapPool_OrderBy {
...
+  Address = 'address',
+  AllowedPools = 'allowedPools',
+  Decimals = 'decimals',
+  Id = 'id',
+  Name = 'name',
+  PriceEth = 'priceEth',
+  PriceUsd = 'priceUsd',
+  Symbol = 'symbol'
+}
+
```
Comment (anxolin) [inline src/api/cow-subgraph/graphql.ts:2794]: All these seems unrelated, although we can relax it and add it in this PR already

**Review by anxolin — APPROVED** [review]
The other altearnative was to add it in the doc (not in metadata), as an optional string. But I'm also good with this approach

**Comment by anxolin** [comment]
To my last comment, I wonder if we forsee any change to environement to make it worth it to version it and have it as a metadata instead of part of the doc itself. Not strong opinion here

**Comment by alfetopito** [comment]
Implemented changes as suggested by Anxo

File: src/schemas/appData.schema.json [hunk 3]
```diff
@@ -20,6 +20,13 @@
       "title": "App Code",
       "type": "string"
     },
+    "environment": {
+      "$id": "#/properties/environment",
+      "description": "Environment in which the order came from",
```
Comment (alfetopito) [inline src/schemas/appData.schema.json:25]:
```suggestion
      "description": "Environment from which the order came from",
```
